### PR TITLE
[JSON-RPC][video] Support double precision in bookmarks

### PR DESCRIPTION
**Pull request by mediaminister**
## Description

This adds double precision for `position` and `total` timings when getting bookmarks from Kodi database with JSON-RPC `Files.GetFileDetails`.

## Motivation and Context

Double precision timings for bookmarks are needed so bookmarks can be synced more precisely with the corresponding API's of online streaming services like Netflix. Currently, when getting bookmark `position` and `total` timings from Kodi database with `Files.GetFileDetails`, the fraction after the comma is not returned properly. The timings are stored in the database with double precision, but returned without the fractional part of the timings.
For instance `"resume":{"position":581.0,"total":2740.0}` is returned while Kodi database contains `581.106487` and `2740.16`

## How Has This Been Tested?

Tested with Kodi master branch on Linux x64 with VRT NU Add-on executing the following JSON-RPC commands:
```
curl -X POST -H "content-type:application/json" http://localhost:8080/jsonrpc -d '{"jsonrpc": "2.0", "method": "Files.SetFileDetails", "params": {"file": "plugin://plugin.video.vrt.nu/play/id/vid-6fec4cf9-ccc5-4c59-8066-c1090eff599b/pbs-pub-e6cd657c-8497-48d8-bd4d-33929ae43ac3", "media": "video", "resume": {"position":1094.5521, "total": 2499.554888}}, "id": "1"}'

curl -X POST -H "content-type:application/json" http://localhost:8080/jsonrpc -d '{"jsonrpc": "2.0", "method": "Files.GetFileDetails", "params": {"file": "plugin://plugin.video.vrt.nu/play/id/vid-6fec4cf9-ccc5-4c59-8066-c1090eff599b/pbs-pub-e6cd657c-8497-48d8-bd4d-33929ae43ac3", "media": "video", "properties": ["resume"]}, "id": "1"}'
```

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/utils/XMLUtils.cpp [hunk 1]
```diff
@@ -306,6 +306,12 @@ TiXmlNode* XMLUtils::SetFloat(TiXmlNode* pRootNode, const char *strTag, float va
   return SetString(pRootNode, strTag, strValue);
 }
 
+TiXmlNode* XMLUtils::SetDouble(TiXmlNode* pRootNode, const char *strTag, double value)
```
Comment (Rechi) [inline xbmc/utils/XMLUtils.cpp]: > Left-align `*` and `&` to the base type they modify.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types

File: xbmc/utils/XMLUtils.h [hunk 2]
```diff
@@ -82,6 +82,7 @@ class XMLUtils
   static void SetStringArray(TiXmlNode* pRootNode, const char *strTag, const std::vector<std::string>& arrayValue);
   static TiXmlNode* SetInt(TiXmlNode* pRootNode, const char *strTag, int value);
   static TiXmlNode* SetFloat(TiXmlNode* pRootNode, const char *strTag, float value);
+  static TiXmlNode* SetDouble(TiXmlNode* pRootNode, const char *strTag, double value);
```
Comment (Rechi) [inline xbmc/utils/XMLUtils.h]: > Left-align `*` and `&` to the base type they modify.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

**Comment by DaveTBlake** [comment]
All changes to JSON API also require a change in version.txt to alert API consumers that things have changed

Unfortunately there is an historic mis-match between how `video.resume` is defined. We have in types.json
```
 "Video.Resume": {
    "type": "object",
    "properties": {
      "position": { "type": "number", "minimum": 0 },
      "total": { "type": "number", "minimum": 0 }
    },
```
yet position and total properties  were implemented as integer. This means that from using the API many API consumers will be expecting integer values, and the change to double may break things. 

Normally a change to internal implementation requires only a patch bump, but given the impact this correction may have it is probably best to bump minor version.

This PR also has a number of cosmetic changes mixed in which make it harder to read what is actually changed. Please remove the format changes made to code lines that you are not editing in any other way.

**Comment by mediaminister** [comment]
I removed the mixed cosmetic changes and bumped the JSON-RPC API version.

**Review by DaveTBlake — APPROVED** [review]
Thanks @mediaminister

**Pull request merged**